tests/test_state_client.cpp

```diff
@@ -42,10 +43,23 @@ TEST_F(smTests, testSMSetStateOff)
 
 TEST_F(smTests, testSMSetStateUndefined)
 {
+    ara::exec::MockStateClient MyStateClient;
+    mySM.stateClient = &MyStateClient;
```

Comment:
this should be moved to constructor - before SM starts operating.

Reply:
Done